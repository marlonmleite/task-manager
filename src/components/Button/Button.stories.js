import React from 'react'
import { storiesOf } from '@storybook/react'
import Button from './'

const storie = storiesOf('Button', module)

storie.add('primary', () => <Button color="primary">My Button</Button>)
