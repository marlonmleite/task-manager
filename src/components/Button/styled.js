import styled from 'styled-components'

export const ButtonStyled = styled.button`
  padding: 0.4375rem 0.75rem;
  line-height: 1.25rem;
  font-size: 0.875rem;
  background-color: ${({ theme, color }) => theme.colors[color]};
  color: ${({ theme, color }) => theme.button.text[color]};
  display: inline-block;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  user-select: none;
  border: 1px solid ${({ theme, color }) => theme.colors[color]};
  border-radius: 2px;
  position: relative;

  :focus,
  :hover {
    text-decoration: none;
    background-image: linear-gradient(transparent, rgba(0,0,0,.05) 40%, rgba(0,0,0,.1));
  }

  :focus {
    outline: none;
  }

  .disabled,
  :disabled {
    opacity: .65;
    box-shadow: none;
  }

  :not([disabled]):not(.disabled) {
    cursor: pointer;
  }

  ${({ loading }) => loading && `
    > span {
      visibility: hidden;
    }
  `};
`

export const LoadingContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`
