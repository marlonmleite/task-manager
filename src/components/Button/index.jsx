import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import { Loading } from 'components/Loading'
import { ButtonStyled, LoadingContainer } from './styled'

class Button extends PureComponent {

  static defaultProps = {
    color: 'primary',
    type: 'button',
    onClick: null,
    loading: false,
  }

  static propTypes = {
    children: PropTypes.node.isRequired,
    onClick: PropTypes.func,
    color: PropTypes.string,
    type: PropTypes.string,
    loading: PropTypes.bool,
  }

  render() {
    const { children, ...props } = this.props

    return (
      <ButtonStyled {...props}>
        <span>{children}</span>
        {props.loading && (
          <LoadingContainer>
            <Loading size="14px" color="secondary" />
          </LoadingContainer>
        )}
      </ButtonStyled>
    )
  }

}

export default Button
