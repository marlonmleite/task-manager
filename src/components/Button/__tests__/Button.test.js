import React from 'react'
import { shallow } from 'enzyme'
import Button from '../'

describe('Button', () => {
  it('should render Button', () => {
    const btn = shallow(<Button onClick={jest.fn()}>Ok</Button>)

    expect(btn).toMatchSnapshot()
  })

  it('should render with loading', () => {
    const btn = shallow(<Button onClick={jest.fn()} loading>Ok</Button>)

    expect(btn).toMatchSnapshot()
  })

  it('should handle onClick', () => {
    const onClick = jest.fn()
    const btn = shallow(<Button onClick={onClick}>Ok</Button>)

    btn.simulate('click')

    expect(onClick).toHaveBeenCalled()
  })
})
