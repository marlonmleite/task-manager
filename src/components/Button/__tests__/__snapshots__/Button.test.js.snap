// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<styled.button
  color="primary"
  loading={false}
  onClick={[MockFunction]}
  type="button"
>
  <span>
    Ok
  </span>
</styled.button>
`;

exports[`Button should render with loading 1`] = `
<styled.button
  color="primary"
  loading={true}
  onClick={[MockFunction]}
  type="button"
>
  <span>
    Ok
  </span>
  <styled.div>
    <styled.div
      color="secondary"
      size="14px"
    />
  </styled.div>
</styled.button>
`;
